collectors/jiraffe/tasks.py

```diff
@@ -1,60 +1,23 @@
 """
 Celery tasks for the JIRA Collector
 """
-from celery import shared_task
+from celery.schedules import crontab
 from celery.utils.log import get_task_logger
-from django.conf import settings
 
-from config.celery import app
-from osidb.core import set_user_acls
-from osidb.models import Affect
+from collectors.framework.models import collector
+from osidb.models import Tracker
 
-from .constants import (
-    JIRA_MAX_TRIES,
-    JIRA_RATE_LIMIT,
-    JIRA_SOFT_TIME_LIMIT,
-    JIRA_SYNC_INTERVAL,
-    JIRAFFE_AUTO_SYNC,
-)
-from .core import get_affects_to_sync, upsert_trackers
+from .collectors import JiraTrackerCollector
 
 logger = get_task_logger(__name__)
 
 
-@app.task(
-    max_retries=JIRA_MAX_TRIES,
-    soft_time_limit=JIRA_SOFT_TIME_LIMIT,
-    rate_limit=JIRA_RATE_LIMIT,
+@collector(
+    base=JiraTrackerCollector,
+    crontab=crontab(),  # run every minute
+    data_models=[Tracker],
+    depends_on=["collectors.bzimport.tasks.flaw_collector"],
 )
-def process_affect(affect_uuid, groups_read, groups_write):
-    """
-    Fetches an affect and creates or updates any Trackers related to it
-    """
-    # Essentially bypass ACLs, celery workers should be able to read/write any
-    # information in order to fulfill their jobs
-    set_user_acls(
-        settings.PUBLIC_READ_GROUPS
-        + [
-            settings.PUBLIC_WRITE_GROUP,
-            settings.EMBARGO_READ_GROUP,
-            settings.EMBARGO_WRITE_GROUP,
-        ]
-    )
-    affect = Affect.objects.get(uuid=affect_uuid)
-    upsert_trackers(affect)
-
-
-@shared_task
-def jiraffe_sync():
-    """
-    Regularly polls the JIRA API for updated trackers to update them internally
-    """
-    if JIRAFFE_AUTO_SYNC:
-        affect_uuids = get_affects_to_sync(JIRA_SYNC_INTERVAL)
-        if affect_uuids:
-            for affect_uuid in affect_uuids:
-                process_affect.delay(
-                    affect_uuid=affect_uuid,
-                    groups_read=settings.PUBLIC_READ_GROUPS,
-                    groups_write=[settings.PUBLIC_WRITE_GROUP],
-                )
+def jira_tracker_collector(collector_obj):
+    logger.info(f"Collector {collector_obj.name} is running")
+    return collector_obj.collect()
```

Comment:
Correct. Do you think we should have it conditional? If so it would be probably good to have some general mechanism of setting the collectors off with having them on by default.

Reply:
Currently if not configured it would just fail in the backend once a minute and if configured with the wrong environment it would probably not link the stuff and just have separate production flaws and stage trackers. Otherwise it should do no harm as the collectors do not write to the other systems.

If it was for some reason not good to have all collectors on locally I would be for creating some more general mechanism of turning them of as part of the collector framework. Eg. we could have registering the collector conditional at https://github.com/RedHatProductSecurity/osidb/blob/master/collectors/framework/models.py#L542 with something like
```
if get_env("COLLECTOR_{name.upper()}", default="True", is_bool=True):
```
(just an idea, not tested)